src/main/fc/fc_core.c

```diff
@@ -237,11 +245,25 @@ static void updateArmingStatus(void)
 
         /* CHECK: Arming switch */
         if (!isUsingSticksForArming()) {
-            // If arming is disabled and the ARM switch is on
-            if (isArmingDisabled() && IS_RC_MODE_ACTIVE(BOXARM)) {
-                ENABLE_ARMING_FLAG(ARMING_DISABLED_ARM_SWITCH);
-            } else if (!IS_RC_MODE_ACTIVE(BOXARM)) {
+            static bool ArmSwitchToggle=false;
+            // If ARM switch is Off or Other Disarm Flags Set
+            if ( isArmingDisabled() || !IS_RC_MODE_ACTIVE(BOXARM) ) { //kbi
+              ENABLE_ARMING_FLAG(ARMING_DISABLED_ARM_SWITCH);
```

Comment:
I don't understand this logic. The `isArmingDisabled() || !IS_RC_MODE_ACTIVE(BOXARM)` check will disable arming for "ARM SWITCH" reason when the switch is in fact not activated.